extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -22,6 +22,7 @@ const QueueEntry = require('../utils/QueueEntry');
 const CredentialsManager = require('../../../credentials/CredentialsManager');
 
 const MPU_CONC_LIMIT = 10;
+const CONSUMER_FETCH_MAX_BYTES = 5000020; // a little over 5MB
```

Comment:
Could you explain why we need a little over 5MB? I think it deserves a comment especially since 5MB looks like a magic value here.